src/db.c

```diff
@@ -1843,7 +1847,10 @@ int doesCommandHaveChannels(struct redisCommand *cmd) {
 /* The base case is to use the keys position as given in the command table
  * (firstkey, lastkey, step).
  * This function works only on command with the legacy_range_key_spec,
- * all other commands should be handled by getkeys_proc. */
+ * all other commands should be handled by getkeys_proc. 
+ * 
+ * NOTE: This function does not guarantee populating the flags for 
+ * the keys, in order to get flags you should use getKeysUsingKeySpecs. */
```

Comment:
i didn't see fixes to the bugs i found in getKeysUsingKeySpecs, or changes in the logic of when getKeysFromCommand uses key-specs, vs when it uses the getkeys_proc.
i guess that's still pending?
p.s. when you edit that, take extra care when thinking about modules.

Reply:
I wasn't planning on updating getKeysFromCommand for this PR, did we want to do that?

Reply:
i guess not.. if we're using `getKeysFromCommandWithSpecs` directly in ACL.
maybe we need to come up with some plan though to test that both of these are returning the same set of keys.
IIRC, Guy wrote some temp code in processCommand to try calling both and assert that they return the same (the run the full test suite and see it doesn't crash).
maybe it's a good idea to repeat that test, but also it would be nice to think of some test that can be added to the test suite.